lib/chef/knife/bootstrap/templates/chef-full.erb

```diff
@@ -1,14 +1,33 @@
-bash -c '
+sh -c '
 <%= "export https_proxy=\"#{knife_config[:bootstrap_proxy]}\"" if knife_config[:bootstrap_proxy] -%>
 
-distro=`uname -s`
+if test "x$TMPDIR" = "x"; then
+  tmp="/tmp"
+else
+  tmp=$TMPDIR
+fi
+
+# secure-ish temp dir creation without having mktemp available (DDoS-able but not exploitable)
+tmp_dir="$tmp/install.sh.$$"
+(umask 077 && mkdir $tmp_dir) || exit 1
 
-if test "x$distro" = "xSunOS"; then
-  if test -d "/usr/sfw/bin"; then
-    PATH=/usr/sfw/bin:$PATH
-    export PATH
+# @param $1 the omnibus root directory
+# @param $2 the requested version of omnibus package
```

Comment:
below we pass latest_current_version_string in as the second parameter, but looking in ```lib/chef/knife/core/bootstrap_context.rb``` it's a string to be passed to install.sh, not just a version. 

Likely values of that would be:

* "-p"
* -"v #{knife_config[:bootstrap_version}"
* -"v #{knife_config[:bootstrap_version} -p"
* "-v #{Chef::VERSION.split(".").first}"